### ISSUE TITLE
failed sync for redis-operator to v0.18.0 :  "redis.redis.redis.opstreelabs.in" is invalid: metadata.annotations: Too long

### ISSUE THREAD

**Issue by RameshPSIT** [OT-CONTAINER-KIT/helm-charts#218]
**Does this issue reproduce with the latest release?**
Yes

**What did you do?**
upgrading from 0.15.3  to 0.18.0 

**What did you see instead?**

`Failed sync attempt to 2fd0cc00eaca0bfc0b1eb247744e068fb2379007: one or more objects failed to apply, reason: [CustomResourceDefinition.apiextensions.k8s.io](https://customresourcedefinition.apiextensions.k8s.io/) "[redis.redis.redis.opstreelabs.in](https://redis.redis.redis.opstreelabs.in/)" is invalid: metadata.annotations: Too long: must have at most 262144 bytes,CustomResourceDefinition.apiextensions.k8s.io "[redisclusters.redis.redis.opstreelabs.in](https://redisclusters.redis.redis.opstreelabs.in/)" is invalid: metadata.annotations: Too long: must have at most 262144 bytes,CustomResourceDefinition.apiextensions.k8s.io "[redisreplications.redis.redis.opstreelabs.in](https://redisreplications.redis.redis.opstreelabs.in/)" is invalid: metadata.annotations: Too long: must have at most 262144 bytes`

**Comment by drivebyer** [comment]
try this https://github.com/argoproj/argo-cd/issues/5704#issuecomment-792337798 @EdagPSIT

**Comment by zumbuschms** [comment]
I have encountered the same issue upgrading the CRDs from the 0.18.1 helm chart. The workaround you suggested does not work. The resource definitions are already defined and trying to create them just gives an error message that the defintions already exist. A following kubectl apply -f ./crds/redis.yaml results in the same error as reported by RameshPSIT.

It seems that kubectl apply tries to write the whole existing definition into the last-applied-configure annotation. This looks like the complete resource definition which ist way over 240K with all the comments.

I switched to kubectl replace to upgrade the CRDs:

`kubectl replace -R -f .\redis-operator\crds`
